test/scWBTC.t.sol

```diff
@@ -0,0 +1,389 @@
+        uint256 initialBalance = 30e8;
+        deal(address(wbtc), address(vault), initialBalance);
+
+        bytes[] memory callData = new bytes[](2);
+        callData[0] = abi.encodeWithSelector(scCrossAssetYieldVault.supply.selector, aaveV3Adapter.id(), initialBalance);
+        callData[1] = abi.encodeWithSelector(scCrossAssetYieldVault.borrow.selector, aaveV3Adapter.id(), 200 ether);
+
+        vault.rebalance(callData);
+
+        // simulate 50% profit
+        uint256 wethInvested = weth.balanceOf(address(wethVault));
+        deal(address(weth), address(wethVault), wethInvested.mulWadUp(1.5e18));
+
+        // assertEq(vault.getProfit(), 100 ether, "profit");
+
+        uint256 totalBefore = vault.totalAssets();
+
+        vault.exitAllPositions(0);
+
+        assertApproxEqRel(wbtc.balanceOf(address(vault)), totalBefore, 0.005e18, "vault wbtc balance");
+        assertEq(vault.totalCollateral(), 0, "vault collateral");
+        assertEq(vault.totalDebt(), 0, "vault debt");
+        assertEq(weth.balanceOf(address(vault)), 0, "weth balance");
+        assertEq(vault.targetTokenInvestedAmount(), 0, "weth invested");
+    }
+
+    /////////////////////////////////// INTERNAL METHODS ////////////////////
+
+    function _deployAndSetUpVault() internal {
+        priceConverter = new WbtcWethPriceConverter();
+        swapper = new WbtcWethSwapper();
+
+        vault = new scWBTC(address(this), keeper, wethVault, priceConverter, swapper);
+
+        vault.addAdapter(aaveV3Adapter);
+
+        // set vault eth balance to zero
+        vm.deal(address(vault), 0);
+        // set float percentage to 0 for most tests
+        vault.setFloatPercentage(0);
+        // assign keeper role to deployer
+        vault.grantRole(vault.KEEPER_ROLE(), address(this));
+    }
+
+    function _assertCollateralAndDebt(uint256 _protocolId, uint256 _expectedCollateral, uint256 _expectedDebt)
+        internal
+    {
+        uint256 collateral = vault.getCollateral(_protocolId);
+        uint256 debt = vault.getDebt(_protocolId);
+        string memory protocolName = _protocolIdToString(_protocolId);
+
+        assertApproxEqAbs(collateral, _expectedCollateral, 1, string(abi.encodePacked("collateral on ", protocolName)));
+        assertApproxEqAbs(debt, _expectedDebt, 1, string(abi.encodePacked("debt on ", protocolName)));
+    }
+
+    function _protocolIdToString(uint256 _protocolId) public view returns (string memory) {
+        if (_protocolId == aaveV3Adapter.id()) {
+            return "Aave V3";
+        }
+
+        revert("unknown protocol");
+    }
+
+    function _deposit(address _user, uint256 _amount) internal returns (uint256 shares) {
+        deal(address(wbtc), _user, _amount);
+
+        vm.startPrank(_user);
+        wbtc.safeApprove(address(vault), type(uint256).max);
+        shares = vault.deposit(_amount, _user);
+        vm.stopPrank();
+    }
```

Comment:
Many of these tests are not really needed, since they are simply copied from other test files. They are essentially testing the functionality of the scCrossAssetYieldVault contract, which has already been extensively tested through tests for scUSDCv2, scUSDT, etc. Instead, focus on testing the added code only, ie price converter and swapper contracts. If these two are working as expected, we can be sure the rest is working as well. That said, there is only value in keeping one happy-path test here just for a sanity check, that will confirm we can deposit/get profits/withdraw from the scWbtc.

Reply:
I removed a few tests, mostly the fuzz tests , will that be enough?